### ISSUE TITLE
MIDI Editor Navigation: Restrict To Time Selection?

### ISSUE THREAD

**Issue by MatejGolian** [jcsteh/osara#1063]
Hello,

I'm curious what you think, but I would personally find it useful if Osara offered a toggle that would optionally restrict MIDI note/CC navigation/selection to only operate within the time selection if one is made. With no time selection, things could operate like now - even with the toggle turned on.

My scenario is this.
I often edit MIDI CCs just under selected notes and remembering what positions a given note starts/ends at can get tricky. For that reason it's quite easy to move past the note you want and edit a CC you don't actually want to edit, because it 'belongs' to another note. If there was a mode to restrict navigation/event selection to only work within the time selection boundaries, it would speed up the workflow, since one wouldn't have to remember note start/end positions and one could be sure that they won't accidentally edit something that they don't want.

So my use case is primarily with MIDI CC events, but perhaps there are other scenarios where something like this could come in useful.

**Comment by ScottChesworth** [comment]
Yeah I see how that would be useful. We'd need a way to distinguish between users reaching the beginning or end of a time selection as opposed to the beginning or end of all MIDI data in the item.
Perhaps when users are at those boundaries and they try to go to previous/next chord, OSARA reports "start of time selection" and "end of time selection". I think that report in combination with not hearing a repeated note preview should explain the cursor isn't moving, and why that's the case.

**Comment by MatejGolian** [comment]
@ScottChesworth glad you find it useful.
To me your suggestion sounds fine. Of course something similar would have to be done with CC message navigation.
